lib/cusolver/libcusolver.jl

```diff
@@ -6206,12 +6224,12 @@ end
     initialize_context()
     @ccall libcusolver.cusolverSpCcsrcholDiag(handle::cusolverSpHandle_t,
                                               info::csrcholInfo_t,
-                                              diag::CuPtr{Cfloat})::cusolverStatus_t
+                                              diag::CuPtr{cuComplex})::cusolverStatus_t
```

Comment:
@amontoison The CUDA headers use simple float/double here, but shouldn't it be complex float/double as I did here? The C/Z bits would indicate so.

Reply:
Hi @maleadt, when you perform a Cholesky of a complex matrix, the matrix must be Hermitian (A = A') and it means that the diagonal of A is real.
The diagonal of the factor `L` of the Cholesky decomposition must be also real such that `LL' = A`.

Reply:
Thanks!